src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -703,4 +737,32 @@ private void commitTransaction(Connection connection) throws SQLException {
         connection.commit();
     }
 
+    private void handleSqlException(String message, SQLException ex)
+            throws InterruptedException, DatabaseOperationException {
+
+        if (ex.getCause() instanceof InterruptedException) {
+            logger.warn("Flight DAO operation interrupted: " + message, ex);
+            String causeText = (ex.getCause() == null) ? "unknown cause" : ex.getCause().toString();
+            throw new InterruptedException(causeText);
+        }
+        logger.error("Flight DAO operation error: " + message, ex);
+        throw new DatabaseOperationException(message, ex);
+    }
+
+    private boolean retrySqlException(SQLException ex)  {
+        final String ss = ex.getSQLState();
+        if (ss.equals("40001") || ss.equals("40P01")) {
+            // Serialization or deadlock
+            logger.info("Caught SQL serialization error - retrying");
+            return true;
+        }
+        if (ss.startsWith("08") || ss.startsWith("53")) {
+            // Connection or resource issue
+            logger.info("Caught SQL connection or resource error - retrying");
+            return true;
+        }
```

Comment:
I don't know if there's a Java/JDBC/Spring SQLState enum (brief googling didn't find me anything). Using an "official" enum would be ideal, but barring that could you move these strings into static variables or an enum (of our own)?

Reply:
I did not find one either. I will make some constants for these magic numbers.

Reply:
It is a bummer. Some of the PSQL status codes are there, but not the ones we care about :(
I defined some constants.